test/phoenix_presence_test.dart

```diff
@@ -0,0 +1,388 @@
+import 'dart:async';
+
+import 'package:phoenix_wings/src/phoenix_push.dart';
+import 'package:phoenix_wings/src/phoenix_socket.dart';
+@TestOn("vm")
+
+import 'package:test/test.dart';
+import 'package:phoenix_wings/src/phoenix_channel.dart';
+import 'package:phoenix_wings/src/phoenix_presence.dart';
+
+MockChannel _mockChannel;
+
+void main() {
+  Map<String, Map<String, dynamic>> presences;
+
+  setUp(() async {
+    _mockChannel = MockChannel();
+
+    presences = {
+      'u1': {'metas': [{'id': 1, 'phx_ref': '1'}]},
+      'u2': {'metas': [{'id': 2, 'phx_ref': '2'}]},
+      'u3': {'metas': [{'id': 3, 'phx_ref': '3'}]}
+    };
+  });
+
+  tearDown(() async {
+  });
+
+  group("Presence construciton", () {
+    test("Returns presence with defaults", () {
+      var phoenixPresence = PhoenixPresence(_mockChannel);
```

Comment:
Almost all of the `var`'s in the test file could be `final`, which is a nice clue for future devs.  Consider replacing all `var` with `final` where they are not being reassigned.